internal/pkg/term/selector/local_selector.go

```diff
@@ -219,6 +274,52 @@ func (s *localFileSelector) listDockerfiles() ([]string, error) {
 	return dockerfiles, nil
 }
 
+// listDirsAndFiles returns the list of directories and files within the current
+// working directory and two subdirectory levels below.
+func (s *localFileSelector) listDirsAndFiles() ([]string, error) {
+	wdDirsAndFiles, err := s.fs.ReadDir(s.workingDirAbs)
+	if err != nil {
+		return nil, fmt.Errorf("read directory: %w", err)
+	}
+	var filesAndDirs []string
+	for _, firstLevel := range wdDirsAndFiles {
+		name := firstLevel.Name()
+		if strings.HasPrefix(name, ".") || name == "copilot" {
+			continue
+		}
+		filesAndDirs = append(filesAndDirs, filepath.Dir(name)+"/"+name)
+		if firstLevel.IsDir() {
+			subFiles, err := s.fs.ReadDir(name)
```

Comment:
i wonder if we could refactor `listDirsAndFiles()` so that we could recursively call the same function twice instead of duplicating logic? something like:
```
listDirsAndFiles(depth int) ([]string, error) {
  var out []string
  for wdDirsAndFiles {
    if depth < 3 && isDir {
       out = append(listDirsAndFiles(depth+1))
    }
  }
}
```
(very pseudocode, hopefully the idea makes sense!)

Reply:
done